core/src/package_store.rs

```diff
@@ -0,0 +1,559 @@
+        } else {
+            Err(errors)
+        }
+    }
+
+    // This function was introduced to avoid repeated code. It takes a map and a package, and adds
+    // all transforms in that package which either exists in the list (include_entries=true) or
+    // doesn't exist in the list (include_entries=false) into the map
+    fn insert_transforms(
+        map: &mut HashMap<String, TransformVariant>,
+        pkg: &Package,
+        include_entries: bool,
+        include_list: &[String],
+    ) -> Result<(), CoreError> {
+        for transform @ Transform {
+            from,
+            to,
+            description: _,
+            arguments: _,
+        } in &pkg.info.transforms
+        {
+            if include_entries == include_list.contains(from) {
+                for output_format in to {
+                    if map
+                        .get(from)
+                        .and_then(|t| t.find_transform_to(output_format))
+                        .is_some()
+                    {
+                        return Err(CoreError::OccupiedTransform(
+                            from.clone(),
+                            output_format.to_string(),
+                            pkg.info.name.clone(),
+                        ));
+                    }
+                    let mut target = map
+                        .remove(from)
+                        .unwrap_or_else(|| TransformVariant::External(HashMap::new()));
+                    target.insert_into_external(
+                        output_format.clone(),
+                        (transform.clone(), pkg.clone()),
+                    );
+                    // Add the modified entry back to the map
+                    map.insert(from.clone(), target);
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    pub fn is_missing_packages(&self) -> bool {
+        !self.awaited_packages.is_empty()
+    }
+
+    pub(crate) fn resolve_request(&mut self, request: PackageID, response: Vec<u8>) {
+        if self.awaited_packages.remove(&request) {
+            self.new_packages.insert(request, response);
+        }
+    }
+
+    pub(crate) fn reject_request<E>(&mut self, request: PackageID, response: E)
+    where
+        E: Error + Send + 'static,
+    {
+        if self.awaited_packages.remove(&request) {
+            self.failed_packages
+                .push(CoreError::Resolve(request.name.clone(), Box::new(response)));
+        }
+    }
+}
+
+impl From<&str> for PackageID {
+    fn from(s: &str) -> Self {
+        #[inline]
+        fn prefix<'a, T>(s: &'a str, prefix: &'static str, t: T) -> Option<(&'a str, T)> {
+            s.starts_with(prefix)
+                .then(|| (s.split_at(prefix.len()).1, t))
+        }
+
+        let (name, target) = None
+            .or(prefix(s, "pkg:", PackageSource::Registry))
+            .or(prefix(s, "pkgs:", PackageSource::Registry))
+            .or(prefix(s, "prelude:", PackageSource::Standard))
```

Comment:
I don't think that we should bloat the language with multiple aliases for the same thing since it might confuse people.

Reply:
I thought `pkgs:` was kinda not-perfekt since we want to say more like "import package abc" not "packages abc", so that's why I added `pkg:` as an alias. I added both `prelude:` and `std:` since it is in the prelude and you are hiding it from there (in many languages there are notions of standard libraries which aren't in the prelude), and `std:` since it is shorter and maybe easier to use? What do you prefer?

Reply:
Realised that our reasoning behind the specifiers might not be that clear. Created #203, that way we can deal with this issue later : )